### PR TITLE
[release/6.0] Update build and test machines

### PR DESCRIPTION
**Pull request by dougbu**
- stop building w/ VS2017 in CI
    - version of VisualStudio is out of support
- use 1ES machines for official builds
- use Ubuntu 18.04 Helix queue under Ubuntu Docker image

nits:
- move to latest Ubuntu Docker image
- remove irrelevant richnav YAML
    - does not build for this branch

### CODE REVIEW

**Review by smitpatel — APPROVED** [review]

**Review by wtgodbe — APPROVED** [review]

File: azure-pipelines.yml [hunk 1]
```diff
@@ -141,10 +141,14 @@ stages:
             timeoutInMinutes: 120
             enablePublishTestResults: true
             pool:
-              vmImage: ubuntu-18.04
+              - ${{ if or(ne(variables['System.TeamProject'], 'internal'), in(variables['Build.Reason'], 'Manual', 'PullRequest', 'Schedule')) }}:
+                vmImage: ubuntu-18.04
+              - ${{ if and(ne(variables['System.TeamProject'], 'internal'), notin(variables['Build.Reason'], 'Manual', 'PullRequest', 'Schedule')) }}:
```
Comment (dougbu) [inline azure-pipelines.yml:146]: typo
```suggestion
              - ${{ if and(eq(variables['System.TeamProject'], 'internal'), notin(variables['Build.Reason'], 'Manual', 'PullRequest', 'Schedule')) }}:
```

**Comment by dougbu** [comment]
I reset this branch and pulled over bb7c910644aa and e2f46c7e179a to correct object syntax and align `$(_CosmosConnectionUrl)` decisions

**Comment by dougbu** [comment]
Rebased on `brecon/60cookie` to confirm continued success

**Comment by dougbu** [comment]
This PR and #39683 appear redundant. Since my `rebase` didn't kick off validation here, closing this one.

**Comment by dougbu** [comment]
Never mind, was completely confused between efcore and aspnetcore☹️

**Pull request merged**